tree/readspeed/src/ReadSpeedCLI.cxx

```diff
@@ -0,0 +1,256 @@
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#include "ReadSpeedCLI.hxx"
+
+#include <ROOT/TTreeProcessorMT.hxx> // for TTreeProcessorMT::SetTasksPerWorkerHint
+
+#include <iostream>
+#include <cstring>
+
+using namespace ReadSpeed;
+
+const auto usageText = "Usage:\n"
+                       " rootreadspeed --files fname1 [fname2 ...]\n"
+                       "               --trees tname1 [tname2 ...]\n"
+                       "               (--all-branches | --branches bname1 [bname2 ...] | --branches-regex bregex1 "
+                       "[bregex2 ...])\n"
+                       "               [--threads nthreads]\n"
+                       "               [--tasks-per-worker ntasks]\n"
+                       " rootreadspeed (--help|-h)\n"
+                       " \n"
+                       " Use -h for usage help, --help for detailed information.\n";
+
+const auto argUsageText =
+   "Arguments:\n"
+   " Specifying files and trees:\n"
+   "   --files fname1 [fname2...]\n"
+   "    The list of root files to read from.\n"
+   "\n"
+   "   --trees tname1 [tname2...]\n"
+   "    The list of trees to read from the files. If only one tree is provided then it will\n"
+   "    be used for all files. If multiple trees are specified, each tree is read from the\n"
+   "    respective file.\n"
+   "\n"
+   "\n"
+   " Specifying branches:\n"
+   "  Branches can be specified using one of the following flags. Currently only one can be used\n"
+   "  at a time.\n"
+   "   --all-branches\n"
+   "    Reads every branch from the specified files and trees.\n"
+   "\n"
+   "   --branches bname1 [bname2...]\n"
+   "    Reads the branches with matching names. Will error if any of the branches are not found.\n"
+   "\n"
+   "   --branches-regex bregex1 [bregex2 ...]\n"
+   "    Reads any branches with a name matching the provided regex. Will error if any provided\n"
+   "    regex does not match at least one branch.\n"
+   "\n"
+   "\n"
+   " Meta arguments:\n"
+   "   --threads nthreads\n"
+   "    The number of threads to use for file reading. Will automatically cap to the number of\n"
+   "    available threads on the machine.\n"
+   "\n"
+   "   --tasks-per-worker ntasks\n"
+   "    The number of tasks to generate for each worker thread when using multithreading.\n";
+
+const auto fullUsageText =
+   "Description:\n"
+   " rootreadspeed is a tool used to help identify bottlenecks in root analysis programs\n"
+   " by providing an idea of what throughput you can expect when reading ROOT files in\n"
+   " certain configurations.\n"
+   " \n"
+   " It does this by providing information about the number of bytes read from your files,\n"
+   " how long this takes, and the differnet throughputs in MB/s, both in total and per thread.\n"
```

Comment:
Maybe give the docs a spell-checker pass :grimacing: 

```suggestion
   " how long this takes, and the different throughputs in MB/s, both in total and per thread.\n"
```

Reply:
Woops, knew I'd mess up somewhere, spell checked the lot now